docs/content/oauth-oidc.md

```diff
@@ -0,0 +1,131 @@
+
+## JSON Web Tokens
+
+To deal with this, OAuth2 established a basic protocol for obtaining _access tokens_, which would be issued by the authorization server and passed around between services in order to identify and authorize the caller. Using opaque tokens however poses many of the same problems as those faced when passing around user identifiers - rather than asking a database for more information the authorization server would instead have to be consulted whenever additional data was needed. While the OAuth2 specification leaves it to the implementation to decide on the format of the tokens, JSON Web Tokens (JWTs) have emerged as the most common one. Using JWTs as tokens include a number of benefits.
+
+* Self-contained - JWTs contain most, or all, of the details needed to identify the caller.
+* Signed - JWTs are signed by the issuer, meaning their authenticity can be verified by the receiver.
+* Immutable - a JWT can't be changed anywhere in the request chain without invalidating the signature.
+
+The OpenID Connect standard took it one step further, and made JWTs _the_ format for tokens, encoding user identity in a new token type called _ID tokens_. This, along with many other enhancements added a a much welcome identity layer on top of the quite minimal OAuth2 standard. For the purpose of this document, we will deal with both access tokens and ID tokens as JWTs.
+
+## Token issuers, discovery and metadata
+
+Any JWT issued by an OAuth2 or OpenID Connect authorization server will carry the URL of the issuer as part of the `iss` claim in the token payload. This value _must_ be matched against the expected issuer(s) when verifying the token. OpenID Connect standardized on requiring the issuer to be a valid HTTPS URL pointing to the address of the issuer. This URL can be used for discovery purposes, obtaining information from the issuer, like available endpoints, that would otherwise need to be kept in applications and policies. The [metadata endpoint](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata) of any compliant OpenID Connect server is found by appending the `/.well-known/openid-configuration` endpoint to the issuer URL. In order to reduce the number of configuration options coded into our policies, we may query this URL and use the endpoints provided in the response for further interactions with the authorization server.
+
+```live:oidc:module
+package oidc
+
+# Error handling and JWT verification omitted for brevity
```

Comment:
Nit: can we omit this comment in this example as well as the ones below?

Reply:
Alright, removed.